### ISSUE TITLE
[BUG] Example http-echo-tinygo doesn't work locally

### ISSUE THREAD

**Issue by FireRB** [wasmCloud/wasmCloud#1177]
**Describe the bug**
By following the instructions on https://github.com/wasmCloud/wasmCloud/blob/main/examples/golang/actors/http-echo-tinygo/README.md, the application doesn't work and got the following error:
```
2023-12-13T22:25:55.246106Z ERROR wasmcloud_host::wasmbus: failed to handle request origin=WasmCloudEntity { public_key: "VAG3QITQQ2ODAOWB5TTQSDJ53XK3SHBEIFNK4AYJ5RKAX2UNSCAPHA5M", link_name: "default", contract_id: "wasmcloud:httpserver" } target=WasmCloudEntity { public_key: "MCFQPBUAOPQVG4ISXCX3IPOKPMJAYTQGDN37C3EBUIMKLDI3JZERYNH2", link_name: "", contract_id: "" } operation="HttpServer.HandleRequest" invocation_id="ee002cfec5d24c5d97b243467610646e" e=failed to handle invocation

Caused by:
    0: failed to instantiate `wasi:http/incoming-handler`
    1: failed to instantiate `wasi:http/incoming-handler` interface
    2: failed to instantiate instance (no bindings satisfied exports):
       
       `wasmcloud:bus/guest` error: import `wasi:clocks/wall-clock@0.2.0-rc-2023-11-10` has the wrong type
       
       Caused by:
           0: instance export `now` has the wrong type
           1: expected func found nothing
       
       `wasi:command/command` error: import `wasi:clocks/wall-clock@0.2.0-rc-2023-11-10` has the wrong type
       
       Caused by:
           0: instance export `now` has the wrong type
           1: expected func found nothing
```

**To Reproduce**
1. Install the dependencies listed on the [readme](https://github.com/wasmCloud/wasmCloud/blob/main/examples/golang/actors/http-echo-tinygo/README.md) file.
1. `git clone https://github.com/wasmCloud/wasmCloud.git`
1. `cd wasmCloud/examples/golang/actors/http-echo-tinygo`
1. `wash build`
2. Replace the image location in wadm.yaml with the output from the previous command.
3. `WASMCLOUD_VERSION=v0.80.0 wash up --nats-websocket-port 4001`
4. `wash app deploy wadm.yaml`
5. Visit localhost:8081 in a browser.

**Expected behavior**
The echo service should respond HTTP 200.

**Screenshots**
![Screenshot 2023-12-13 at 14 28 46](https://github.com/wasmCloud/wasmCloud/assets/62574417/a654b8c6-2596-4609-9945-20e365786a3e)

**Environment (please complete the following information) **
 - OS: MacOS
 - Shell: zsh
 - wasmcloud Version: 0.80.0
 - wash version: 0.24.1
 - tinygo version: 0.30.0

**Comment by vados-cosmonic** [comment]
Hey @FireRB thanks for the report -- I've started to take a look at it, and it looks like this might be a case of the ecosystem mvoing quickly! The WIT definitions that the golang example echo code look to be outdated, with what is loaded into the host. 

In addition, the existing golang code was written in a pre-resources world, so it looks like there is a bit of updating that needs to happen.

**Comment by vados-cosmonic** [comment]
Hi @FireRB, it turns I can't reproduce this failure -- would you mind sharing how you installed `wash` ?

**Comment by FireRB** [comment]
I also have go installed earlier. Not sure whether that conflict with tinygo.
```
go version go1.21.3 darwin/arm64
```

**Comment by vados-cosmonic** [comment]
**tl;dr Newer upstream code seems to have snuck into our macOS binary build of `wash`, please install w/ the Rust toolchain**.

Thanks so much for your detailed response @FireRB, it was invaluable in narrowing down the problem  -- it looks like this problem is caused by the difference in installing via `brew` versus `cargo install`!

There is a bug in the current tap (and more pertinently upstream in our building of the binary that is delivered by `brew`). 

I tried a couple things to get here:
- ensuring the version of `wit-bindgen-go` matching (it does, this version of `wash` uses [`wit-bindgen-go`](https://github.com/bytecodealliance/wit-bindgen/tree/main/crates/go)) v0.13.1 under the covers) -- you can confirm this if you look at `gen/echo.c`
- installing via `brew` vs `cargo`

The ecosystem has moved quite a bit since this example was written, and we've updated to match -- this means that the code on `main` is much newer (and uses newer versions of core dependencies like `wasmtime` and `wit-bindgen` itself), but it looks like one of those newer deps *snuck into the binary release for MacOS*. 

Long story short, unfortunately our binary build (ex. `wash-aarch64-apple-darwin` if you're on an M1/2/3) seems to be broken, but installing `wash` from source w/ the Rust toolchain (i.e. `cargo`) is not.

```console
cargo install wash-cli@0.24.0 --locked
```
